src/hotspot/share/services/diagnosticCommand.hpp

```diff
@@ -395,6 +395,24 @@ class ClassHierarchyDCmd : public DCmdWithParser {
   virtual void execute(DCmdSource source, TRAPS);
 };
 
+class AOTEndTrainingDCmd : public DCmd {
+public:
+  AOTEndTrainingDCmd(outputStream* output, bool heap) : DCmd(output, heap) { }
+    static const char* name() { return "AOT.end_training"; }
+    static const char* description() {
+      return "End AOT training and create the cache.";
+    }
+    static const char* impact() {
+      return "High: [insert reason here]";
+    }
+    static const JavaPermission permission() {
+    JavaPermission p = {"java.lang.management.ManagementPermission",
+                        "monitor", nullptr};
```

Comment:
Doesn't it require control permission instead of monitor permission to end the training run?

Reply:
I'll take a look, I copied that from a similar dump command, but yes I guess we end up terminating the app - good question!